patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -425,6 +425,38 @@ private class HookList
 				g.Instance(item).ModifyResearchSorting(item, ref itemGroup);
 		}
 
+		private delegate bool? DelegateCanResearch(Item item);
+		private static HookList HookCanResearch = AddHook<DelegateCanResearch>(g => g.CanResearch);
+		public static bool? CanResearch(Item item) {
+			if (item.IsAir)
+				return null;
+
+			bool? canResearch = item.modItem?.CanResearch();
+
+			if (canResearch != null)
+				return canResearch;
+
+			foreach (var g in HookCanResearch.arr) {
+				canResearch = g.Instance(item).CanResearch(item);
+				
+				if (canResearch != null)
+					return canResearch;
+			}
```

Comment:
all hooks like this need to decide on a precedence behaviour. Returning the result of the first hook to give a non-null (vanilla) response will be weird when one mod says yes and another says no.

You need to decide, and describe in the documentation of the hook, which should take priority.

Do this by suggesting use cases for these hooks.